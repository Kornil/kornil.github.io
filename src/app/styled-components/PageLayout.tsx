import styled from "app/styled-components";

const PageLayout = styled.div`
  height: inherit;
  background-color: #fff;
  color: #111;
  padding: 0 18px;
  grid-template-areas:
    "nav nav"
    "main  main";
  grid-template-rows: 50px 1fr;
`;

export default PageLayout;
