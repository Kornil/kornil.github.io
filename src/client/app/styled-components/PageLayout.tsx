import styled from "react-emotion";

import { lightTextColor, textColor } from "./styled-constants";

const PageLayout = styled.div`
  min-height: 100%;
  padding: 0;
  margin: 0;
  background-color: #dfdfdf;
  font-family: "Helvetica Neue", "Calibri Light", Roboto, sans-serif;
  color: ${textColor};
  text-align: center;
  display: grid;
  grid-template-rows: 50px 1fr 50px;
  grid-template-columns: 0.3fr 1fr 0.3fr;

  @media (max-width: 425px) {
    grid-template-columns: 1fr;
  }

  > * {
    grid-column: 2;
  }

  /* fix scrollbar issue */
  padding-left: calc(100vw - 100%);

  /* default em style */
  em {
    margin: 0;
    color: ${lightTextColor};
  }

  /* default link style */
  a {
    color: #3d3d3d;
    text-decoration: none;

    :hover {
      color: #1f1f1f;
    }
  }

  /* default h styles*/
  h1 {
    margin: 0 0 16px;
    font-size: 2em;
  }

  h2 {
    color: ${lightTextColor};
    font-size: 1.5em;
    margin: 0 0 32px;
    font-weight: normal;
  }

  h3 {
    font-size: 1.3em;
  }

  p,
  li {
    font-size: 1.1em;
    line-height: 1.58;
  }

  li {
    margin: 8px 0;
  }
`;

export default PageLayout;
