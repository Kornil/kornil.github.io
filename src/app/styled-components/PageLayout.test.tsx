import { shallow } from "enzyme";
import React from "react";

import PageLayout from "./PageLayout";

test("it works", () => {
  const wrapper = shallow(<PageLayout />);
  expect(wrapper).toMatchSnapshot();
});
