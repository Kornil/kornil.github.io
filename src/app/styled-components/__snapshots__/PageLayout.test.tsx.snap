// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.emotion-0 {
  height: inherit;
  background-color: #fff;
  color: #111;
  grid-template-areas: "nav nav" "main  main";
  grid-template-rows: 50px 1fr;
}

<div
  className="emotion-0 emotion-1"
/>
`;
